testsuite/smoke-test/src/consensus/quorum_store_fault_tolerance.rs

```diff
@@ -77,46 +117,58 @@ async fn test_onchain_config_quorum_store_enabled() {
         swarm.root_key(),
         swarm.chain_info().root_account().address(),
     );
-
     let rest_client = swarm.validators().next().unwrap().rest_client();
-    let current_consensus_config: OnChainConsensusConfig = bcs::from_bytes(
-        &rest_client
-            .get_account_resource_bcs::<Vec<u8>>(
-                CORE_CODE_ADDRESS,
-                "0x1::consensus_config::ConsensusConfig",
-            )
-            .await
-            .unwrap()
-            .into_inner(),
-    )
-    .unwrap();
 
+    let current_consensus_config = get_current_consensus_config(&rest_client).await;
     let inner = match current_consensus_config {
         OnChainConsensusConfig::V1(inner) => inner,
         OnChainConsensusConfig::V2(_) => panic!("Unexpected V2 config"),
     };
-
     // Change to V2
     let new_consensus_config = OnChainConsensusConfig::V2(ConsensusConfigV1 { ..inner });
+    update_consensus_config(&cli, root_cli_index, new_consensus_config).await;
 
-    let update_consensus_config_script = format!(
-        r#"
-    script {{
-        use aptos_framework::aptos_governance;
-        use aptos_framework::consensus_config;
-        fun main(core_resources: &signer) {{
-            let framework_signer = aptos_governance::get_signer_testnet_only(core_resources, @0000000000000000000000000000000000000000000000000000000000000001);
-            let config_bytes = {};
-            consensus_config::set(&framework_signer, config_bytes);
-        }}
-    }}
-    "#,
-        generate_blob(&bcs::to_bytes(&new_consensus_config).unwrap())
-    );
-    cli.run_script(root_cli_index, &update_consensus_config_script)
+    generate_traffic_and_assert_committed(&mut swarm, &validator_peer_ids).await;
+}
+
+// TODO: remove when quorum store becomes the in-code default
+/// Test in case of disaster whether we can roll back to non-quorum store
+#[tokio::test]
+async fn test_onchain_config_quorum_store_disabled() {
```

Comment:
Should we enable and then disable, so we know we can actually rollback rather than having it in the genesis? Not sure if that makes any difference.

Reply:
we had a forge test previously to roll a feature back and forth 100 times and see if it still survives

Reply:
Changed this to cycle through on/off 5 times. Please re-review! :)